Build failures are not passed through

When `yarn build` encounters build errors in a project, the exit status will still be `0`.

This can be observed with the example projects in this repo:
```shell
rm .yarn/local-build-cache.json # in case it exists
╰─ yarn build && echo $?          
➤ YN0000: ┌ Building All
➤ YN0009: │ ┌ Errors for packages/examples/words/sit
➤ YN0009: │ │ src/index.ts:1:14 - error TS1005: ';' expected.
➤ YN0009: │ │ 1 ftygukjhu +c asda
➤ YN0009: │ │                ~~~~
➤ YN0009: │ │ Found 1 error.
➤ YN0009: │ └ End packages/examples/words/sit
➤ YN0000: └ Build finished with 2 errors
➤ YN0000: [11:2/13]
➤ YN0000: Done in 11s 899ms
0
```

As we can see, the exit status is `0` and the build failure will not fail consecutive processes.

--- a/packages/plugins/plugin-build/src/commands/build/index.ts
+++ b/packages/plugins/plugin-build/src/commands/build/index.ts
@@ -2,6 +2,7 @@ import { BaseCommand } from "@yarnpkg/cli";
 import {
   Configuration,
   Manifest,
+  MessageName,
   Project,
   StreamReport,
   Workspace,
@@ -165,7 +166,10 @@ export default class Build extends BaseCommand {
         }
 
         // build all the things
-        await supervisor.run();
+        const ranWithoutErrors = await supervisor.run();
+        if (ranWithoutErrors === false) {
+          report.reportError(MessageName.BUILD_FAILED, "Build failed");
+        }
       }
     );
 

--- a/packages/plugins/plugin-build/src/commands/build/supervisor.ts
+++ b/packages/plugins/plugin-build/src/commands/build/supervisor.ts
@@ -603,6 +603,8 @@ class BuildSupervisor {
 
     // commit the build log
     await this.saveBuildLog();
+
+    return this.buildReport.failCount === 0;
   };
 
   // This is a very simple requestAnimationFrame polyfil

